### ISSUE TITLE
Merging instanced node from another scene bug

### ISSUE THREAD

**Issue by SoulForMachine** [godotengine/godot#31744]
**Godot version:**
v3.1.1.stable.official
v3.2.dev.custom_build.e71cf83f

**OS/device including version:**
All

**Issue description:**
When using Merge From Scene, and then selecting a node in another scene that is an instantiation of a third scene, and then checking Editable Children on the merged node, all children of that node will be duplicated.

**Steps to reproduce:**
- Create a scene, add root Node2D, name it Scene1, add a Sprite node as child. Save as Scene1.tscn.
- Create a scene, add root Node2D, name it Scene2, instantiate Scene1.tscn as child. Save as Scene2.tscn.
- Create a scene, add root Node2D, name it Scene3.
- On root node Scene3, right-click and use Merge From Scene, select Scene2.tscn, select Scene1 node.
- On newly merged node, right-click and check Editable Children.
- Select the grayed Sprite node move it a little bit.
- Save the scene as Scene3.tscn and run it.
- The scene in game has 2 sprites displayed.
- Close the game and close the Scene3.tscn tab in editor.
- Reopen Scene3.tscn again. Now 2 Sprite nodes are visible in editor too, one is grayed and one isn't. When one is selected the other is automatically selected too.

**Minimal reproduction project:**
[MergeInstanceTest.zip](https://github.com/godotengine/godot/files/3552074/MergeInstanceTest.zip)

**Comment by SoulForMachine** [comment]
I think I found out what the problem is. `EditorSubScene`, from which a sub-tree is selected to merge, did not instance the source scene with `PackedScene::GEN_EDIT_STATE_INSTANCE`, so if the node we are merging is in fact an instanced scene in the source scene, it won't have its `data.instance_state` set. Later, when Editable Children is checked for the merged node and the scene is saved, `SceneState::_parse_node` will treat child Sprite of the merged node as not being part of the instanced scene and will write it into the tscn.

```
[gd_scene load_steps=3 format=2]

[ext_resource path="res://Scene1.tscn" type="PackedScene" id=1]
[ext_resource path="res://icon.png" type="Texture" id=2]

[node name="Scene3" type="Node2D"]

[node name="Scene1" parent="." instance=ExtResource( 1 )]

[node name="Sprite" type="Sprite" parent="Scene1"]
position = Vector2( 160, 31 )
texture = ExtResource( 2 )

[editable path="Scene1"] 
```

So the sprite is duplicated. I changed the scene instantiation call in `EditorSubScene::_path_changed` to:
`scene = ps->instance(PackedScene::GEN_EDIT_STATE_INSTANCE);`

This fixes the problem, but some more feedback would be helpful since I'm new to Godot. I see that @reduz and @akien-mga touched most of this function, so please advise if this is correct solution.

**Comment by KoBeWi** [comment]
Still valid in ee36e08

**Comment by akien-mga** [comment]
Fixed by #34892 and #45943.
